pages/index.tsx

```diff
@@ -13,43 +13,90 @@ import Layout from "../components/Layout";
 import EventModal from "../components/EventModal";
 import styles from "../styles/events.module.css";
 import { IEventDTO } from "../dtos";
-import { reduceOpacity, defaultColors, getEvents } from "../utils";
+import { reduceOpacity, defaultColors } from "../utils";
 import { SubjectColor } from "../types";
 
-import { google } from "googleapis";
+const localizer = momentLocalizer(moment);
 
-export interface IFormatedEvent {
-  title: string;
-  place: string;
-  link: string;
-  start: Date;
-  end: Date;
-  groupId: number;
-  filterId: number;
+const edKey = "eventData";
+const luKey = "lastUpdateEvents";
+
+// Fetch event data using the API
+async function getData(): Promise<IEventDTO[]> {
+  // ! CHANGE TO: const domain = process.env.NEXT_PUBLIC_DOMAIN; BEFORE DEPLOYING
+  const domain = window.location.origin;
```

Comment:
```suggestion
      const domain = process.env.NEXT_PUBLIC_DOMAIN || window.location.origin;
```

Reply:
NEXT_PUBLIC_DOMAIN is "localhost:3000" when developing and "calendario.cesium.di.uminho.pt" when deployed. Since deploy previews are neither of those, I entered "window.location.origin". Since window.location.origin always gets the correct link, we could just leave it like that. But what you did doesn't work because NEXT_PUBLIC_DOMAIN also exists locally on `.env.local`, just with a different value.

Reply:
Based on your comment , this line doesn't work only in deploy previews, and if so i think its ok.

Reply:
I think it should just be `window.location.origin`, that way it works for all cases. Having that "or" is unnecessary as NEXT_PUBLIC_DOMAIN will always have a value, so in your code having `process.env.NEXT_PUBLIC_DOMAIN || window.location.origin` is the same as having `process.env.NEXT_PUBLIC_DOMAIN`, meaning that `window.location.origin` will never be the value of `domain`.

Reply:
Does `window.location.origin` works in production?
If so you can leave it as it is

Reply:
It depends on where the page is rendered. On the server, there is no such thing as `window`.
But we don't need that, we can use `process.env.NEXT_PUBLIC_DOMAIN` (as Diogo stated) and stick with the env files.

Reply:
@ruilopesm @MarioRodrigues10 apparently we don't even need to specify the domain, can can just do `fetch("/api/...")`... I've tested and it works :sweat_smile: Resolved!